src/main/java/org/myrobotlab/service/Pcf8574.java

```diff
@@ -54,57 +54,44 @@ public class Pcf8574 extends Service
   public class Publisher extends Thread {
 
     public Publisher(String name) {
-      super(String.format("%s.publisher", name));
+      super(String.format("%s.poller", name));
     }
 
-    // FIXME - have a read "rate" in Hz
     void publishPinData() {
       // Read a single byte containing all 8 pins
-      read8();
-      PinData[] pinArray = new PinData[pinDataCnt];
-      for (int i = 0; i < pinArray.length; ++i) {
-        PinData pinData = new PinData(i, read(i));
-        pinArray[i] = pinData;
-        String address = pinData.pin;
-
-        // handle individual pins
-        if (pinListeners.containsKey(address)) {
-          List<PinListener> list = pinListeners.get(address);
-          for (int j = 0; j < list.size(); ++j) {
-            PinListener pinListner = list.get(j);
-            if (pinListner.isLocal()) {
-              pinListner.onPin(pinData);
-            } else {
-              invoke("publishPin", pinData);
-            }
-          }
+      readRegister();
+      List<PinData> pinArray = new ArrayList<>();
+      // reads a byte of data in
+      readRegister();

```

Comment:
Why the readRegister() at line 65 when there is one at line 62?

Reply:
excellent catch, removed one !